spyder/preferences/languageserver.py

```diff
@@ -103,126 +103,132 @@ class LSPServerEditor(QDialog):
+            '<tt>{port}</tt> in the server arguments field to '
+            'automatically fill in the respective values.<br>'
+        )
         self.parent = parent
         self.external = external
-        bbox = QDialogButtonBox(QDialogButtonBox.Ok | QDialogButtonBox.Cancel)
-        self.button_ok = bbox.button(QDialogButtonBox.Ok)
-        self.button_cancel = bbox.button(QDialogButtonBox.Cancel)
-        self.button_ok.setEnabled(False)
 
-        description = _('To create a new configuration, '
-                        'you need to select a programming '
-                        'language, along with a executable '
-                        'name for the server to execute '
-                        '(If the instance is local), '
-                        'and the host and port. Finally, '
-                        'you need to provide the '
-                        'arguments that the server accepts. '
-                        'The placeholders <tt>{host}</tt> and '
-                        '<tt>{port}</tt> refer to the host '
-                        'and the port, respectively.')
-        server_settings_description = QLabel(description)
-        server_settings_description.setWordWrap(True)
-
-        lang_label = QLabel(_('Language:'))
+        # Widgets
+        self.server_settings_description = QLabel(description)
+        self.lang_label = QLabel(_('Language:'))
         self.lang_cb = QComboBox(self)
-        self.lang_cb.setToolTip(_('Programming language provided '
-                                  'by the LSP server'))
+        self.external_cb = QCheckBox(_('External server'), self)
+        self.host_label = QLabel(_('Host:'))
+        self.host_input = QLineEdit(self)
+        self.port_label = QLabel(_('Port:'))
+        self.port_spinner = QSpinBox(self)
+        self.cmd_label = QLabel(_('Command to execute:'))
+        self.cmd_input = QLineEdit(self)
+        self.args_label = QLabel(_('Server arguments:'))
+        self.args_input = QLineEdit(self)
+        self.json_label = QLabel(self.JSON_VALID, self)
+        self.conf_label = QLabel(_('LSP Server Configuration:'))
+        self.conf_input = CodeEditor(None)
+
+        self.bbox = QDialogButtonBox(QDialogButtonBox.Ok |
+                                     QDialogButtonBox.Cancel)
+        self.button_ok = self.bbox.button(QDialogButtonBox.Ok)
+        self.button_cancel = self.bbox.button(QDialogButtonBox.Cancel)
+
+        # Widget setup
+        self.setMinimumSize(self.MIN_SIZE)
+        self.setWindowTitle(_('LSP server editor'))
+
+        self.server_settings_description.setWordWrap(True)
+
+        self.lang_cb.setToolTip(
+            _('Programming language provided by the LSP server'))
         self.lang_cb.addItem(_('Select a language'))
-        self.lang_cb.addItems(LSP_LANGUAGES)
+        languages = LSP_LANGUAGES[:]
+        # Even if users add a new Python config there, we will not use it
+        languages.remove('Python')
```

Comment:
It seems you can't remove it from here but you need to do it from the dropdown menu instead.

Reply:
Hmmm, I do not follow

I made a copy of the list, before removing it and passing it the dropdown.

Reply:
```python
        languages = LSP_LANGUAGES[:]
        # Even if users add a new Python config there, we will not use it
        languages.remove('Python')
        self.lang_cb.addItems(languages)
````

Reply:
Ok, that's why I started with *It seems*. Then I don't know why our tests are failing.

Reply:
>Then I don't know why our tests are failing.

🤷‍♂ will take a look